packages/backend-defaults/src/entrypoints/rootHttpRouter/createLifecycleMiddleware.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RootLifecycleService } from '@backstage/backend-plugin-api';
+import { ServiceUnavailableError } from '@backstage/errors';
+import { HumanDuration, durationToMilliseconds } from '@backstage/types';
+import { RequestHandler } from 'express';
+
+export const DEFAULT_TIMEOUT = { seconds: 5 };
+
+/**
+ * Options for {@link createLifecycleMiddleware}.
+ * @public
+ */
+export interface LifecycleMiddlewareOptions {
+  lifecycle: RootLifecycleService;
+  /**
+   * The maximum time that paused requests will wait for the service to start, before returning an error.
+   *
+   * Defaults to 5 seconds.
+   */
+  requestPauseTimeout?: HumanDuration;
```

Comment:
Ah, we didn't discuss the need for this one so far but given that we're adding config here I think it makes sense to expose this too. Iirc there's been some situations where it would be useful for people to be able to bump this a bit.

Perhaps we can make it even more explicit in the naming, thinking something like `startupRequestPauseTimeout`?

Don't forget to document these in `docs/backend-system/` as well! 🙏